### ISSUE TITLE
Vector search fails after merging with a table

### ISSUE THREAD

**Issue by christian-hnz** [lancedb/lance#1905]
I receive a `OSError: Io error: Execution error: Schema error: field summary does not exist in the RecordBatch` when doing a vector search after merging a table **with an extra column** (this is essential), that is, a column that is neither used for merging nor in the vector search.

The behavior described below has been tested (amongst others) with `pylance` 0.9.12 and python 3.11.7 on Ubuntu 22.04.

Here is a minimal example.

```
import random
import shutil
import string

import polars as pl
import pyarrow as pa

import lance

shutil.rmtree("testlance", ignore_errors=True)


def get_string_seq(n: int, length: int = 10) -> list[str]:
    random.seed(a=111)
    return ["".join(random.choices(string.ascii_letters, k=length)) for _ in range(n)]


def get_vec_seq(n: int, length: int = 20) -> list[list[float]]:
    random.seed(a=111)
    return [[random.gauss(0.0, 1.0) for _ in range(length)] for _ in range(n)]


df = pl.DataFrame(
    {
        "text": get_string_seq(1000),
    }
).with_row_index("id_")

vec = pl.DataFrame(
    {
        "summary": get_string_seq(1000),  # this seems to cause the issue
        "vector": get_vec_seq(1000),
    },
    schema_overrides={"vector": pl.Array(pl.Float32, 20)},
).with_row_index("id_")

dataset = lance.write_dataset(df.to_arrow(), "testlance")
dataset.merge(vec.to_arrow(), left_on="id_")
dataset.to_table(
    nearest={
        "column": "vector",
        "q": pa.array([0.0 for _ in range(20)], type=pa.float32()),
        "k": 5,
    }
)
```
The final `to_table` generates the error. If one
* drops the `summary` column from `vec` or
* removes the `nearest` argument form the `to_table`,
then the issue goes away.

**Comment by wjones127** [comment]
Hi @christian-hnz. Thanks for reporting this. I can reproduce this on 0.9.12 (the latest release as of now).

Good new is this is already fixed by #1901. I'll add a unit test for this specific case in #1468 just to be sure it keeps working, though. We'll make a release around Monday that will include this fix.

**Comment by christian-hnz** [comment]
Wow, that was fast. Thanks! Looking forward to the new release :+1:
